### DISCUSSION TITLE
ESP32 Berry - passing mem/var values

### DISCUSSION THREAD

**Discussion by gr10001** [arendst/Tasmota#16117] [Q&A] [answered]
Hi,

I just started moving from rules to Berry and i'd appreciate some help if possible.
My aim is to set a relay on/off base on the value of 2 DS18B20 and values i get store in mem or var. 
I currently get mem values set by mqtt and looking at the documentation i can really find a way to read mem or var other than tasmota.cmd("mem"), which seems to work but then getting the actual value recorded in the specific mem variable seems to be rather tricky.  I've tried multiple combinations of string.find, like string.find(mem,"'Mem2':") but i always get -1. Any advice?
thanks!

**Comment by sfromis** [comment, marked as answer]
The output of `tasmota.cmd` is a map, not a string. Do not try to use string operations. (I've fixed the docs page to point this out, instead of suggesting that output is a string)

You can get a Mem variable using a construct like `tasmota.cmd('Mem2')['Mem2']`

More generally, it is recommended to use the `persist` [module](https://tasmota.github.io/docs/Berry/#persist-module) of Berry instead, for improved persistence.

**Reply by gr10001** [reply]
thanks, it worked but i'll look at the persist module as well
